lib/safe-pg-migrations/plugins/statement_insurer/remove_column_index.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module SafePgMigrations
+  module StatementInsurer
+    module RemoveColumnIndex
+      def remove_column_with_composite_index(table, column)
+        existing_indexes = indexes(table).select { |index|
+          index.columns.size > 1 && index.columns.include?(column.to_s)
+        }
+
+        return unless existing_indexes.any?
+
+        error_message = <<~ERROR
+          Cannot drop column #{column} from table #{table} because composite index: #{existing_indexes.map(&:name).join(', ')} is present
+        ERROR
```

Comment:
Add the fact that if the index is stil necessary without the columns they should "re"create it before droping the ones listed here